code/game/g_combat.c

```diff
@@ -22,6 +22,37 @@ void ScorePlum( gentity_t *ent, vec3_t origin, int score ) {
 	plum->s.time = score;
 }
 
+/*
+============
+DamagePlum
+============
+*/
+void DamagePlum( gentity_t *attacker, vec3_t origin, int damage ) {
+	gentity_t *plum;
+	char userinfo[MAX_INFO_STRING];
+	int damagePlumsEnabled;
+
+	if ( !attacker || !attacker->client ) {
+		return;
+	}
+
+	// check if the client supports damage plums
+	trap_GetUserinfo( attacker->s.number, userinfo, sizeof( userinfo ) );
+	damagePlumsEnabled = atoi( Info_ValueForKey( userinfo, "cg_damagePlums" ) );
+
+	if ( !damagePlumsEnabled ) {
+		return;
+	}
+
+	plum = G_TempEntity( origin, EV_DAMAGEPLUM );
+	// only send this temp entity to the attacker
+	plum->r.svFlags |= SVF_SINGLECLIENT;
+	plum->r.singleClient = attacker->s.number;
+	//
+	plum->s.otherEntityNum = attacker->s.number;
+	plum->s.time = damage;
```

Comment:
I wonder if we really need a server event for this? Can't the client do this not far from where the "hit" sound is played, i.e. `CG_TransitionPlayerState`?

https://github.com/ec-/baseq3a/blob/aa8dac0e13b2c149b6d196882482e8e1429bc8ab/code/cgame/cg_playerstate.c#L483-L527